### PR TITLE
LO: Automatically optimize sets using artifice mods

### PR DESCRIPTION
**Pull request by robojumper**
* Optimizes builds using artifice mods 
grafik
preferring a highest total tier, then stat order (since that's how we sort sets as well).
* Gets rid of the weird TXX (+mods) display since we now just sort by final tier and stats.
* Instead highlights automatically picked mods using a border.

### CODE REVIEW

File: src/app/loadout/loadout-ui/Sockets.m.scss [hunk 1]
```diff
@@ -12,3 +12,9 @@
     --item-size: var(--item-icon-size);
   }
 }
+
+.automaticallyPicked {
+  :global(.item-img) {
+    border: 1px solid coral;
```
Comment (bhollis) [inline src/app/loadout/loadout-ui/Sockets.m.scss:18]: I might suggest `$orange` to try and suggest that it's DIM that picked the mod. Or maybe a green.

**Review by bhollis — APPROVED** [review]

File: src/app/loadout-builder/process-worker/process-utils.ts [hunk 2]
```diff
@@ -193,11 +185,263 @@ export function pickAndAssignSlotIndependentMods(
...
+      cost: subTreeCost,
+      expensive: subTreeExpensive,
+    });
+    statIndex += 1;
+  }
+  return bestResult;
+}
+
+/*@__INLINE__*/
+function isExpensiveMod(statHash: number) {
```
[the remark this answers is not in the document]
Comment (robojumper) [reply src/app/loadout-builder/process-worker/process-utils.ts:430]: "cost" was perhaps the worst possible name, sorry. I've renamed things to make it clear that the "cost to the next tier per stat" has two dimensions, mod costs and needed points.

File: src/app/loadout-builder/process-worker/process-utils.ts [hunk 3]
```diff
@@ -193,11 +185,263 @@ export function pickAndAssignSlotIndependentMods(
...
+ * If there's an optimistic estimate of how many tiers can be gained by stat mods + artifice mods, this also aborts as soon
+ * as a node with this tier total is reached. This gives a small efficiency boost when there aren't many slot-specific mods.
+ */
+function exploreAutoModsSearchTree(
+  info: LoSessionInfo,
+  items: ProcessItem[],
+  /** The base stats from our set + fragments + ... */
+  setStats: number[],
+  /** The stats we have explored in this search process */
+  explorationStats: number[],
```
[the remark this answers is not in the document]
Comment (robojumper) [reply src/app/loadout-builder/process-worker/process-utils.ts:336]: I added some more comments -- does it make sense now?

**Review by nev-r — APPROVED** [review]
working fantastic, plugging everything properly. lets get it merged asap this is useful

**Review by bhollis — APPROVED** [review]

**Pull request merged**